samples/matter/lock/src/app_event.h

```diff
@@ -6,44 +6,42 @@
 
 #pragma once
 
-#include <cstdint>
+#include "event_manager.h"
+
+enum class AppEventType : uint8_t { None = 0, NUSCommand, LockEvent, ThreadWiFiSwitch };
+
+struct AppEvent : public Event {
+	AppEvent()
+		: Event(EventSource::Application, static_cast<uint8_t>(AppEventType::None), nullptr, this,
+			sizeof(*this))
+	{
+	}
+	AppEvent(AppEventType type, EventHandler handler = nullptr)
+		: Event(EventSource::Application, static_cast<uint8_t>(type), handler, this, sizeof(*this))
+	{
+	}
+
+	/**
+	 * @brief Construct a new App Event object from context
+	 *
+	 * User must ensure that context exists, is defined and has the same size as the AppEvent class,
+	 * if the context does not exist The AppEvent will not be updated with any data.
+	 *
+	 * @param context serialized context of the AppEvent
+	 */
+	AppEvent(const void *context)
+	{
+		if (context) {
+			memcpy(this, context, sizeof(*this));
```

Comment:
I'd say this is kind of obscure and may be error prone. For instance, what happens with the context if the event gets destroyed, e.g. it goes out of scope, or the passed context is destroyed?

I don't have a better instant solution, but maybe aforementioned dedicated EventContext class could work.